method_comparison/peft_bench/README.md

```diff
@@ -0,0 +1,216 @@
+
+### Adapter Configuration Example
+
+```json
+{
+    "base_model_name_or_path": "facebook/opt-350m",
+    "bias": "none",
+    "fan_in_fan_out": false,
+    "inference_mode": false,
+    "init_lora_weights": true,
+    "lora_alpha": 16,
+    "lora_dropout": 0.1,
+    "modules_to_save": null,
+    "peft_type": "LORA",
+    "r": 8,
+    "target_modules": ["q_proj", "v_proj"],
+    "task_type": "CAUSAL_LM"
+}
+```
+
+## Prompt Categories
+
+The benchmark automatically runs across all prompt categories for consistent comparison:
+- **short** - Brief prompts (1-2 sentences)
+- **medium** - Moderate length prompts (paragraph-level)
+- **long** - Extended prompts (multiple paragraphs)
+
+Results are tracked separately for each category, allowing analysis of how different PEFT methods perform across varying input lengths.
+
+## Results Structure
+
+Results are saved in a structured JSON format with three main sections:
+
+### `run_info`
+- Execution metadata (timestamp, duration, status)
+- Hardware information (GPU type, CUDA version, etc.)
+- Error information (if applicable)
+- PEFT and benchmark configurations
+
+### `generation_info`
+- Memory usage logs at different stages
+- Per-category metrics (inference time, time per token, etc.)
+- Overall aggregated metrics
+- Individual sample results for detailed analysis
+
+### `meta_info`
+- Model information (ID, PEFT method)
+- Parameter counts (adapter, total, ratio)
+- Model size information (base model, adapter)
+- System and package information
+
+## Key Metrics
+
+### Inference Performance
+- **Inference Time**: Total time for generation per category
+- **Time Per Token**: Normalized time accounting for different generation lengths
+- **Inference Overhead**: Percentage increase compared to base model
+
+### Memory Usage
+- **Peak GPU Memory**: Maximum GPU memory during benchmark
+- **Peak RAM Memory**: Maximum RAM usage
+- **Memory Logs**: Detailed tracking at each stage
+
+### Parameter Efficiency
+- **Adapter Parameters**: Number of parameters in the PEFT adapter
+- **Parameter Ratio**: Percentage of total model parameters that are in the adapter
+- **Adapter Size**: Memory footprint of the adapter in MB
+
+## File Structure
+
+```
+peft_bench/
+├── README.md                        # This documentation
+├── run.py                          # Main benchmark runner
+├── utils.py                        # Core utilities and data structures
+├── data.py                         # Data preparation and prompt handling
+├── default_benchmark_params.json   # Default benchmark configuration
+├── configs/
+│   ├── prompts.json                # Benchmark prompts by category
+│   └── sample_config.json          # Example configuration
+├── experiments/
+│   └── lora/
+│       ├── lora_r8/                # LoRA rank 8 experiment
+│       └── lora_r16/               # LoRA rank 16 experiment
+└── results/                        # Benchmark results (auto-generated)
+```
+
+## Adding New Experiments
```

Comment:
This section can IMO also removed, especially if the `### Adapter Configuration Example` is adjusted according to my proposal.